From a0e5bb3ea70ddc2bd52a8c69b84980249cc4aba7 Mon Sep 17 00:00:00 2001
From: David Edmundson <kde@davidedmundson.co.uk>
Date: Wed, 19 Jul 2017 10:03:55 +0100
Subject: Don't perform wayland specific fixes when on X

Summary:
fd2e850156ac7aa9c9dc2cf46652b2a1f1fc3a07 introduces some behaviour
changes that affect X even though they should be only Wayland related.

BUG: 381130

Reviewers: #plasma

Subscribers: graesslin, plasma-devel, #frameworks

Tags: #plasma, #frameworks

Differential Revision: https://phabricator.kde.org/D6776
---
 src/plasmaquick/dialog.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plasmaquick/dialog.cpp b/src/plasmaquick/dialog.cpp
index a1416db..43fc196 100644
--- a/src/plasmaquick/dialog.cpp
+++ b/src/plasmaquick/dialog.cpp
@@ -1157,7 +1157,7 @@ bool Dialog::event(QEvent *event)
     if (event->type() == QEvent::Expose) {
         auto ee = static_cast<QExposeEvent*>(event);
 
-        if (ee->region().isNull()) {
+        if (!KWindowSystem::isPlatformWayland() || ee->region().isNull()) {
             return QQuickWindow::event(event);
         }
 
-- 
cgit v0.11.2
